jetty-server/src/main/java/org/eclipse/jetty/server/session/DefaultSessionCache.java

```diff
@@ -37,16 +38,26 @@
     /**
      * The cache of sessions in a hashmap
      */
-    protected ConcurrentHashMap<String, Session> _sessions = new ConcurrentHashMap<>();
+    private final ConcurrentMap<String, Session> _sessions;
 
     private final CounterStatistic _stats = new CounterStatistic();
 
     /**
      * @param manager The SessionHandler related to this SessionCache
      */
     public DefaultSessionCache(SessionHandler manager)
+    {
+        this(manager, new ConcurrentHashMap<>());
+    }
+
+    /**
+     * @param manager The SessionHandler related to this SessionCache
+     * @param sessionMap The session map implementation to use
+     */
+    public DefaultSessionCache(SessionHandler manager, ConcurrentMap<String, Session> sessionMap)
```

Comment:
other option to avoid null issues is:
```suggestion
    {
        this(manager, null);
    }

    /**
     * @param manager The SessionHandler related to this SessionCache
     * @param sessionMap The session map implementation to use
     */
    protected DefaultSessionCache(SessionHandler manager, ConcurrentMap<String, Session> sessionMap)
    {
        super(manager);
        _sessions = sessionMap == null ? new ConcurrentHashMap() : sessionMap;
```

Reply:
Also make this constructor protected